### DISCUSSION TITLE
[Enhancement] Nested c# files

### DISCUSSION THREAD

**Discussion by luismts** [dotnet/maui#91] [General]
# Summary (edited)

> I think this is part of visual studio tools. But I leave it here so that it can influence the VS team internally there. 

Now that we are seeing some c# approaches, I think that is time for the nested c# files with partial classes. 

->MyView.cs
-->MyView.logic.cs

Just to split the UI and the UI logic. A lot of people would love that.

**Comment by mrwcjoughin** [comment]
you can already (and have always been able to) do this with partial classes - absolutely nothing stopping you

**Comment by GiampaoloGabba** [comment]
yepp partial classes are here just for that :)
Plus some IDEs provide the ability to configure file nesting in solution explorer as you want! Maybe there is a plugin for VS/VSC, i haven't checked that. But with JetBrains Rider is super easy :)

**Comment by GiampaoloGabba** [comment]
oh, also possible duplicate of #83

**Comment by charlesroddie** [comment]
You can split up .net code into multiple files, and these files will get compiled when you build a project. In any language, not just those with the concept of partial classes.

**Comment by luismts** [comment]
Yes, that was what I mean. I was talking about nested files by default or at least having the option to make that happens. 

I just used the wrong word. Thanks for the comments guys.

cc @GiampaoloGabba @charlesroddie @mrwcjoughin

**Comment by charlesroddie** [comment]
@luismts the right place for a request for nested C# files is https://github.com/dotnet/csharplang .

**Comment by Happypig375** [comment]
This is a tooling request, not a language request. File at https://github.com/dotnet/roslyn instead.
